### PR TITLE
Provide an article on chess - Chess Vocab: Speaking the common tongue ( issue #29 )

### LINKED ISSUE #29: Provide an article on chess - Chess Vocab: Speaking the common tongue
**Is your feature request related to a problem? Please describe.** :bulb:
This article is for the learning section of pbchess.

**Describe the solution you'd like** :sparkles:
- Provide a list of all possible chess-related words in alphabetical order along with one or two-line descriptions for each.

**Describe alternatives you've considered** :star2:
NA

**Additional context** :memo:
NA

### PR DESCRIPTION
**Pull request by Arshanshagarwal**
I have added a list of all chess-related words in alphabetical order along with one or two-line descriptions for each. I also made an indexing system to access each Alphabetic group easily.

Screenshots: 

Learn

learn 2

### CODE REVIEW

File: client/src/components/learn/Terms.json [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+      "title": "Opening variation",
+      "meaning": "A line in a particular opening. For example, in the Sicilian defence there are many different lines which are called variations."
+    },
+    {
+      "title": "Outpost",
+      "meaning": "A position on the chess board where a piece is close to the enemy but difficult for the opponent to remove. There is no reason it has to be a knight, nor does it have to be protected by a pawn."
+    },
+    {
+      "title": "Overworked/Overloaded",
+      "meaning": "Overworked, also known as overloaded, is a tactical weakness in chess where a piece is tasked to defend multiple threats at the same time. In many cases, this results in the overworked piece being unable to move and/or being forced to prioritize which task it can preform in order to mitigate loses."
```
Comment (adi-g15) [inline client/src/components/learn/Terms.json:281]: Typo: ~preform~ _perform_

**Review by adi-g15 — APPROVED** [review]
Looks Good to me 👍

@fave77

**Review by adi-g15 — APPROVED** [review]
image

These aren't errors, but can be fixed @Arshanshagarwal , this is causing the Codacy action to fail>
As for codecov ignore it for now, it's since more code has been added, and existing tests not covering it.

**Review by fave77 — CHANGES_REQUESTED** [review]
@Arshanshagarwal please make the following changes:
- The styling have unintended effects on home page, like so:
<a href="https://ibb.co/Jx5VVf1">IMG-20210312-014349</a>
<a href="https://ibb.co/KGCmjkz">IMG-20210312-014330</a>
- Do not modify anything inside `Learn` component, except for maybe alignment (if needed)
- Create a `Vocabs` component along with your json, js, css, and just import that inside `Learn` component, like so:
```bash
...
const Learn = _ => {
  return (
    <Vocabs /> // the component should be here, just before Videos
    <Videos />
  );
};
```
- Choose class-names with extra caution, it might be the reason for homepage's styling issue

**Review by fave77 — CHANGES_REQUESTED** [review]
@Arshanshagarwal just do the following and your PR will be merged:
- get rid of the `placeholder-img` selector in both `vocab.css` and `learn.css`
- always name the folder of a component in a lower-case format, so `Vocab` to `vocab` (there's no right way, but since the codebase adheres to that, so follow this for the sake of consistency)
- speaking of consistency, export the component at the last line, not while declaring it
- lastly, make the learn component `functional` since it is stateless
```bash
...
const Learn = _ => {
  return (
    <Vocabs />
    <Videos />
  )
};

export default Learn;
```

- (Optional) - provide smooth scrolling when someone clicks an alpahbet

File: client/src/components/learn/learn.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+import Videos from "../videos/videos";
+import Vocab from "../vocab/vocab";
+import "./learn.css";
 
-const Learn=() => {
+const Learn = () => {
   return (
-    <Videos />
-  )
+    <>
```
Comment (fave77) [inline client/src/components/learn/learn.js]: Use a wrapper div

File: client/src/components/vocab/vocab.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React, { Component } from "./../../../node_modules/react";
```
Comment (fave77) [inline client/src/components/vocab/vocab.js]: Import libraries the proper way

**Review by fave77 — CHANGES_REQUESTED** [review]
- `learn` shouldn't contain json file
- please squash your commits going forward as that prompts into a frequent PR update (I'm letting it pass this time, so no need to squash now for this PR)
@Arshanshagarwal

**Review by fave77 — APPROVED** [review]

**Pull request merged**